label_studio_ml/text_extractor/model.py

```diff
@@ -174,3 +185,69 @@ def get_numbers_from_string(string: str) -> float:
     numbers = [number[0].replace(",", ".") for number in numbers]
     numbers = [abs(float(number)) for number in numbers]
     return numbers
+
+
+def get_coordinate_numbers_from_string(string: str) -> tuple[float]:
+    """Extract the first two numbers from a string.
+
+    Supports various notation of coordinates.
+
+    Supported coordinate formats are:
+        - "2'456'435"
+        - "2456435"
+        - "2.456.435"
+        - "2456435"
+        - "2,456,435"
+        - "456'435"
+        - etc.
+
+    Args:
+        string (str): The string to extract the number from.
+
+    Returns:
+        tuple[float]: The extracted numbers.
+    """
+    numbers = re.findall(COORDINATE_ENTRY_REGEX, string)
+    if len(numbers) == 2:
+        print(numbers[0])
+        print(numbers[1])
+        return int("".join(numbers[0])), int("".join(numbers[1]))
+    else:
+        return tuple()
+
+
+def extract_coordinates(result_text: str, rect: fitz.Rect, page_number: int) -> tuple[int]:
+    """Extract coordinates from OCR result.
+
+    Args:
+        result_text (str): The recognized text.
+        rect (fitz.Rect): The bounding box rectangle forthe coordinates.
+        page_number (int): The page number on which the coordinates are to be found.
+
+    Returns:
+        tuple[int]: Coordinates (N, E) as integers.
+    """
+    try:
+        east, north = get_coordinate_numbers_from_string(result_text)
+    except ValueError:
+        print("Could not recognize any coordinates.")
+        return None
+    # Note, the coordinate class deals with confusion of east and north automatically.
+    # This is true for Switzerland. For other countries, this might not be the case.
+
+    if east > 1e6 and north > 1e6:
+        coordinate = LV95Coordinate(
+            CoordinateEntry(east),
+            CoordinateEntry(north),
+            rect=rect,
+            page=page_number
+        )
+    else:
+        coordinate = LV03Coordinate(
+            CoordinateEntry(east),
+            CoordinateEntry(north),
+            rect=rect,
+            page=page_number
+        )
```

Comment:
Can we reuse the Coordinate.from_values() method here?

Reply:
Yes we can.